### ISSUE TITLE
Remove `src/interpreter/interpreter-non-det.ts` (currently not working anyway)

### ISSUE THREAD

**Issue by martin-henz** [source-academy/js-slang#1695]
Remove all of `src/interpreter` once https://github.com/source-academy/sicp/issues/990 is resolved.

Currently, Source §3 Non-Det is not working. Older version works. E.g. the first example in SICP JS 4.3
https://sourceacademy.nus.edu.sg/sicpjs/4.3.1
works without error (but with spacing issues). (currently deployment of SA@NUS of today)

The newest version in SA.org gives a runtime error:
https://share.sourceacademy.org/s9irv
gives:

main.13f449e4.js:2 RangeError: Maximum call stack size exceeded
This is regardless whether the amb operator is used.

display(1);
gives the same error. Source §3 Non-Det does not work at all, currently.

**Comment by RichDom2185** [comment]
Is this a duplicate of #1476?

**Comment by martin-henz** [comment]
> Is this a duplicate of #1476?

No, it's in addition to #1476. I've clarified the title of this issue.

**Comment by leeyi45** [comment]
Resolved by #1725
